### DISCUSSION TITLE
API Request Rate Limiting

### DISCUSSION THREAD

**Discussion by margaretjoanmiller** [emacs-twist/twist.nix#148] [Q&A]
Hi! I was just wondering if it's on the roadmap to update twist such that it doesn't make so many API requests to github? I have an api key set up which helps some but I still get rate limited. Using a vpn is my current workaround but that doesn't seem ideal.

**Comment by akirak** [comment]
@0x4d6165 Hello, thank you for evaluating twist.

Yes, it is normal to hit the API rate limit of GitHub. You can prevent it by generating a personal access token on [Settings](https://github.com/settings/apps) and add it to your `~/.config/nix/nix.conf`:

```
access-tokens = github.com=ghp_SOME_ENCODED_STRING
```

The details are unofficially [documented here](https://github.com/emacs-twist/twist.nix/blob/8dd0feb03b33b697baec51063ae0e81536eaa171/doc/emacs-twist.org#add-an-access-token-to-github). I didn't further work on the documentation because I didn't have enough confidence with the API. It seems that I need to put more resources on this project, as some people are beginning to adopt it (or at least trying it out).

**Comment by akirak** [comment]
This discussion has been converted from an issue, as it's not a bug of twist.nix but rather about its usage and configuration.
